File: MangaST_1_11/src/main/java/com/example/mangast/manga/MangaRequest.java
package com.example.mangast.manga;


import com.example.mangast.manga.categories.Category;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;


import java.util.List;

@Builder
@Getter
@Setter
public class MangaRequest {
        @NotNull(message = "100")
        @NotEmpty(message = "100")
        private String title;

        @NotNull(message = "101")
        @NotEmpty(message = "101")
        private String authorName;


        private TypeManga typeManga;

        @NotNull(message = "102")
        @NotEmpty(message = "102")
        private String description;

        @NotNull(message = "103")
        @Size(min = 1, message = "Choice at least one category")
        private List<Category> mangaCategories;

        private MangaStatus status;

}

File: MangaST_1_11/FrontMangaST/src/app/services/models/manga-request.ts
/* tslint:disable */
/* eslint-disable */
import { Category } from '../models/category';
export interface MangaRequest {
  authorName: string;
  description: string;
  mangaCategories: Array<Category>;
  status?: 'ANNOUNCE' | 'ONGOING' | 'PAUSED' | 'COMPLETED' | 'DISCONTINUED';
  title: string;
  typeManga?: 'MANGA' | 'MANHWA';
}
